lua/telescope/pickers.lua

```diff
@@ -491,6 +491,44 @@ function Picker:hide_preview()
   -- 2. Resize prompt & results windows accordingly
 end
 
+-- A simple interface to remove an entry from the results window without closing
+-- telescope.
+-- @param delete_cb function
+function Picker:delete_selection(delete_cb)
+  local original_selection_strategy = self.selection_strategy
+  self.selection_strategy = "row"
+  delete_cb = (delete_cb and type(delete_cb) == "function") and delete_cb or function() end
+
+  local delete_selections = self._multi:get()
+  if vim.tbl_isempty(delete_selections) then
+    table.insert(delete_selections, self:get_selection())
+  end
+
+  local selection_index = {}
+  for result_index, result_entry in ipairs(self.finder.results) do
+    if vim.tbl_contains(delete_selections, result_entry) then
+      table.insert(selection_index, result_index)
+    end
+  end
+
+  -- Sort in reverse order as removing an entry from the table shifts down the
+  -- other elements to close the hole.
+  table.sort(selection_index, function(x, y) return x > y end)
+  for _, index in ipairs(selection_index) do
+    local selection = table.remove(self.finder.results, index)
+    delete_cb(selection)
+  end
+
+  if #delete_selections > 1 then
```

Comment:
What if the multi selection was only one? `self._multi:get()` actually returned something and we need to create a new one, right? Maybe just set a flag if you insert into `delete_selections`? Might be wrong tho

Reply:
Ah yes, that is correct. Thanks for pointing it out. I will use a flag to see whether we used `multi_selection` or not and reset it accordingly.